code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -179,101 +135,120 @@
 		else
 			icon_state = "[base_icon][get_bot_flag(bot_mode_flags, BOT_MODE_ON)]"
 
+/mob/living/simple_animal/bot/cleanbot/vv_edit_var(var_name, var_value)
+	. = ..()
+	if(var_name == NAMEOF(src, base_icon))
+		update_appearance(UPDATE_ICON)
+
+/mob/living/simple_animal/bot/cleanbot/proc/deputize(obj/item/knife, mob/user)
+	if(!in_range(src, user) || !user.transferItemToLoc(knife, src))
+		balloon_alert(user, "couldn't attach!")
+		return FALSE
+	balloon_alert(user, "attached [knife]!")
+	weapon = knife
+	if(!(bot_cover_flags & BOT_COVER_EMAGGED))
+		weapon.force = weapon.force / 2
+	add_overlay(image(icon = weapon.lefthand_file, icon_state = weapon.inhand_icon_state))
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/on_entered,
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+	return TRUE
+
+/mob/living/simple_animal/bot/cleanbot/proc/update_titles()
+	name = initial(name) //reset the name
+	ascended = TRUE
+
+	for(var/title in (prefixes + suffixes))
+		for(var/title_name in title)
+			if(!(title_name in stolen_valor))
+				ascended = FALSE
+				continue
+
+			if(title_name in officers_titles)
+				commissioned = TRUE
+			if(title in prefixes)
+				name = title[title_name] + " [name]"
+			if(title in suffixes)
+				name = "[name] " + title[title_name]
+
 /mob/living/simple_animal/bot/cleanbot/bot_reset()
-	..()
-	if(weapon && bot_cover_flags & BOT_COVER_EMAGGED)
-		weapon.force = weapon_orig_force
-	ignore_list = list() //Allows the bot to clean targets it previously ignored due to being unreachable.
+	. = ..()
 	target = null
 
 /mob/living/simple_animal/bot/cleanbot/proc/on_entered(datum/source, atom/movable/AM)
 	SIGNAL_HANDLER
+	if(!weapon || !has_gravity() || !iscarbon(AM))
+		return
 
-	zone_selected = pick(BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
-	if(weapon && has_gravity() && ismob(AM))
-		var/mob/living/carbon/C = AM
-		if(!istype(C))
-			return
-
-		if(!(C.job in stolen_valor))
-			stolen_valor += C.job
+	var/mob/living/carbon/stabbed_carbon = AM
+	if(!(stabbed_carbon.mind.assigned_role.title in stolen_valor))
+		stolen_valor += stabbed_carbon.mind.assigned_role.title
 		update_titles()
 
-		INVOKE_ASYNC(weapon, /obj/item.proc/attack, C, src)
-		C.Knockdown(20)
+	zone_selected = pick(BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	INVOKE_ASYNC(weapon, /obj/item.proc/attack, stabbed_carbon, src)
+	stabbed_carbon.Knockdown(20)
 
 /mob/living/simple_animal/bot/cleanbot/attackby(obj/item/attacking_item, mob/living/user, params)
 	if(istype(attacking_item, /obj/item/knife) && !user.combat_mode)
-		to_chat(user, span_notice("You start attaching \the [attacking_item] to \the [src]..."))
+		balloon_alert(user, "attaching [attacking_item]...")
```

Comment:
Wouldn't this balloon_alert get kinda lengthy if `attacking_item` was a long name? Might be better to just have "attaching..."?

Reply:
it's a knife, so either 'the knife', 'the combat knife', 'the butcher knife', 'the kitchen knife', either way I don't think it's any bad, but I could